accounts/abi/reflect.go

```diff
@@ -24,6 +24,64 @@ import (
 	"strings"
 )
 
+// ToStruct converts a interface of a runtime type into a interface of the
+// given type
+// e.g. turn
+// var fields []reflect.StructField
+// fields = append(fields, reflect.StructField{
+// 		Name: "X",
+//		Type: reflect.TypeOf(new(big.Int)),
+//		Tag:  reflect.StructTag("json:\"" + "x" + "\""),
+// }
+// into
+// type TupleT struct { X *big.Int }
+func ToStruct(in interface{}, proto interface{}) interface{} {
+	inType, protoType := reflect.TypeOf(in), reflect.TypeOf(proto)
+	switch {
+	case inType.ConvertibleTo(protoType):
+		return reflect.ValueOf(in).Convert(protoType).Interface()
+	case inType.Kind() == reflect.Struct:
+		if err := copyStruct(proto, in); err != nil {
+			return nil
+		}
+		return proto
+	case inType.Kind() == reflect.Array:
+		if err := copyArray(proto, in); err != nil {
+			return nil
+		}
+		return proto
+	default:
+		// Use set as a last ditch effort
+		if err := set(reflect.ValueOf(proto), reflect.ValueOf(in)); err != nil {
+			return nil
+		}
+		return proto
+	}
+}
+
+// copyStruct tries to copy each field of in to out.
+func copyStruct(out interface{}, in interface{}) error {
+	valueIn := reflect.ValueOf(in)
+	valueOut := reflect.ValueOf(out).Elem()
+	for i := 0; i < valueOut.NumField(); i++ {
+		if err := set(valueOut.Field(i), valueIn.Field(i)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func copyArray(out interface{}, in interface{}) error {
+	valueIn := reflect.ValueOf(in)
+	valueOut := reflect.ValueOf(out).Elem()
```

Comment:
Should you not also check that the output is an array-like type?

Reply:
You're right will change